internal/kibana/connector_data_source.go

```diff
@@ -0,0 +1,30 @@
+package kibana
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceConnector() *schema.Resource {
+	return &schema.Resource{
+		Description: "Search for a connector by name, space id, and type. Note, that this data source will fail if more than one connector shares the same name.",
+		ReadContext: dataSourceConnectorRead,
+		Schema:      connectorSchema,
+	}
+}
+
+func dataSourceConnectorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	connectorName := d.Get("name").(string)
+	if err := d.Set("name", connectorName); err != nil {
+		return diag.FromErr(err)
+	}
+
+	spaceId := d.Get("space_id").(string)
+	if err := d.Set("space_id", spaceId); err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
Why do we need the `Get/Set` dance here, the set bit feels wrong reading through this.

Reply:
Can you suggest an instance of what it should look like? Was just following the pattern from here: https://github.com/wandergeek/terraform-provider-elasticstack/blob/connector-data-source/internal/kibana/role_data_source.go#L163-L169.

Reply:
That code gets a `name` property and then sets the ID. In this case, you're just getting `name` and then setting `name` on the same resource, as in a noop just writing the same value back to the same place it was read from. 

> of what it should look like?

```
func dataSourceConnectorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
	return resourceConnectorsRead(ctx, d, meta)
}
```

or even just use `resourceConnectorsRead` [here](https://github.com/elastic/terraform-provider-elasticstack/pull/607/files#diff-c42f369829e0231d33f049572ebaf354e9a1d51d1608531b4b6a2f122d6e7f68R13)

Reply:
Sick one.